### PR TITLE
apply effective-dart style to bloc package

### PR DESCRIPTION
**Pull request by felangel**
## Status
**READY**

## Breaking Changes
NO

## Description
- Adhere to [Effective Dart](https://github.com/tenhobi/effective_dart) (#561)

## Todos
- [X] Tests
- [X] Documentation
- [X] Examples

## Impact to Remaining Code Base
None

### CODE REVIEW

**Comment by tenhobi** [comment]
I can see the CI failed 'cause the SDK versions mismatch or something like that.

I am a bit confused, because Travis prints that the Dart sdk has version 2.1.0, but the stable version is around 2.5 now, isn't it?

What can I do to help to pass it?

image

**Comment by felangel** [comment]
@tenhobi its because effective_dart requires dart 2.1.1. Would it be possible for you to change it to require >=2.0.0 < 3.0.0?

**Comment by tenhobi** [comment]
Yea, I copied it from the pedantic package I think, dunno why they require it. I can change it and publish new version, sure.

**Comment by tenhobi** [comment]
Why travis uses that version tho? The stable Dart version is 2.5.2. now.

**Comment by felangel** [comment]
@tenhobi I can just bump up the travis dart version 👍

**Comment by tenhobi** [comment]
@felangel I have released patch version `1.0.1`. https://pub.dev/packages/effective_dart

**Comment by felangel** [comment]
@tenhobi can you review?

File: packages/bloc/lib/src/bloc.dart [hunk 1]
```diff
@@ -4,8 +4,10 @@ import 'package:bloc/bloc.dart';
 import 'package:meta/meta.dart';
 import 'package:rxdart/rxdart.dart';
 
+/// {@template bloc}
```
Comment (tenhobi) [inline packages/bloc/lib/src/bloc.dart]: Wow, this is possible? Super cool!

**Pull request merged**